### PR TITLE
8344190: Cleanup code in sun.net.www.protocol.http and sun.net.www.protocol.https after JEP 486 integration

### PR DESCRIPTION
**Pull request by jaikiran**
Can I please get a review of this code which cleans up usages of SecurityManager and related APIs? This addresses https://bugs.openjdk.org/browse/JDK-8344190.

No new tests have been added and testing of tier1, tier2, tier3 is in progress.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8344190](https://bugs.openjdk.org/browse/JDK-8344190): Cleanup code in sun.net.www.protocol.http and sun.net.www.protocol.https after JEP 486 integration (**Sub-task** - P4)

### Reviewers
 * [Daniel Fuchs](https://openjdk.org/census#dfuchs) (@dfuch - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/22232/head:pull/22232` \
`$ git checkout pull/22232`

Update a local copy of the PR: \
`$ git checkout pull/22232` \
`$ git pull https://git.openjdk.org/jdk.git pull/22232/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 22232`

View PR using the GUI difftool: \
`$ git pr show -t 22232`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/22232.diff">https://git.openjdk.org/jdk/pull/22232.diff</a>

</details>
<details><summary>Using Webrev</summary>

[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/22232#issuecomment-2485228492)
</details>

### CODE REVIEW

File: src/java.base/unix/classes/sun/net/www/protocol/http/ntlm/NTLMAuthentication.java [hunk 1]
```diff
@@ -71,7 +71,6 @@
  */
 
 public class NTLMAuthentication extends AuthenticationInfo {
-    private static final long serialVersionUID = 170L;
```
Comment (dfuch) [inline src/java.base/unix/classes/sun/net/www/protocol/http/ntlm/NTLMAuthentication.java:74]: There is also a windows implementation of this class

**Review by dfuch — COMMENTED** [review]
Pfew! That's a lot of complex code going away! Good work!

A few minor comments. Well, actually, just one. Please make sure to run all tier1,tier2,tier3 tests.

**Review by dfuch — APPROVED** [review]

**Review by dfuch — APPROVED** [review]

**Comment by jaikiran** [comment]
Thank you Daniel for the review.

**Pull request closed without merging**